client/fingerprint/network_unix.go

```diff
@@ -30,21 +30,41 @@ func NewNetworkFingerprinter(logger *log.Logger) Fingerprint {
 	return f
 }
 
+func flagsSet(flags net.Flags, test net.Flags) bool {
```

Comment:
> A minor comment - Can you please add some comments around the logic of the method and add some tests? Also since they are not exported, it would be nice to move them down in the file.

I am not happy to add comments and some tests to one line function, also to leave it away from `Fingerprint()`. So, I tried to include the `flagSet()` an `flagClear()` in Fingerprint(). How about this idea?